### PR TITLE
Osm2mimir configuration

### PR DESCRIPTION
**Pull request by crocme10**

### CODE REVIEW

File: src/settings/osm2mimir.rs [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+                            e
+                        ))
+                    })?,
+                ),
+            );
+        }
+
+        if let Some(nb_admin_replicas) = self.nb_admin_replicas {
+            m.insert(
+                String::from("elasticsearch.admin_replicas"),
```
Comment (amatissart) [inline src/settings/osm2mimir.rs:403]:
```suggestion
                String::from("elasticsearch.admins_replicas"),
```

**Review by amatissart — COMMENTED** [review]
A few inconsistencies need to be fixed so that command line arguments for `osm2mimir` are applied correctly, and everything should be ok.

File: src/osm_reader/osm_store.rs [hunk 2]
```diff
@@ -33,25 +33,31 @@
...
 use osmpbfreader::{OsmId, OsmObj, StoreObjs};
 use slog_scope::info;
 
-#[cfg(not(feature = "db-storage"))]
-use slog_scope::warn;
+#[cfg(feature = "db-storage")]
+use crate::settings::osm2mimir::Database;
+
+// #[cfg(not(feature = "db-storage"))]
+// use slog_scope::warn;
```
[the remark this answers is not in the document]
Comment (crocme10) [reply src/osm_reader/osm_store.rs:45]: done, thanks

File: src/settings/osm2mimir.rs [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+        if let Some(connection_string) = self.connection_string.clone() {
+            m.insert(
+                String::from("elasticsearch.connection_string"),
+                Value::new(None, connection_string),
+            );
+        }
+
+        if let Some(nb_way_shards) = self.nb_street_shards {
+            m.insert(
+                String::from("elasticsearch.way_shards"),
```
[the remark this answers is not in the document]
Comment (crocme10) [reply src/settings/osm2mimir.rs]: :+1:

**Review by amatissart — APPROVED** [review]

**Pull request merged**